bin/internal/update_dart_sdk.sh

```diff
@@ -60,25 +60,51 @@ if [ ! -f "$ENGINE_STAMP" ] || [ "$ENGINE_VERSION" != `cat "$ENGINE_STAMP"` ]; t
   }
 
   # `uname -m` may be running in Rosetta mode, instead query sysctl
-  if [ "$OS" = 'Darwin' ]; then
+  if [ -z "$ARCH" ] && [ "$OS" = 'Darwin' ]; then
     # Allow non-zero exit so we can do control flow
     set +e
-    # -n means only print value, not key
-    QUERY="sysctl -n hw.optional.arm64"
-    # Do not wrap $QUERY in double quotes, otherwise the args will be treated as
-    # part of the command
-    QUERY_RESULT=$($QUERY 2>/dev/null)
-    if [ $? -eq 1 ]; then
```

Comment:
here we were only checking for exit code 1, however there could have been other non-zero exit codes. in particular, if `sysctl` is not on path, bash will return code 127.